common/protobuf.c

```diff
@@ -85,7 +86,10 @@ protobuf_recv_message(int fd, const ProtobufCMessageDescriptor *descriptor)
 	ASSERT(descriptor);
 
 	uint32_t buflen = 0;
-	ssize_t bytes_read = fd_read(fd, (char *)&buflen, sizeof(buflen));
+	ssize_t bytes_read;
+	do {
+		bytes_read = fd_read(fd, (char *)&buflen, sizeof(buflen));
```

Comment:
fd always blocking?

Reply:
this is wat is the patch for, in case of interrupted syscall the message would not be read.